### PR TITLE
ImportC: add Win64 preprocessor support

### PR DESCRIPTION
**Pull request by WalterBright**
Let's see how this fares.

### CODE REVIEW

**Comment by trikko** [comment]
Can you add tcc as preprocessor too? 

It runs on many hosts (i386/x86_64/arm/aarch64/riscv64 <-> Win/Linux/macOS/FreeBSD/NetBSD/OpenBSD) and it works really well with importc since its preprocessed files are simplified. 

Can be compiled as c library too.

**Comment by WalterBright** [comment]
When `checkwhitespace` fails, it no longer tells you which file it failed on.

**Comment by WalterBright** [comment]
The Win64 ImportC tests are failing because the Microsoft preprocessor insists on emitting the filename it is processing, which the test suite then regards as TEST_OUTPUT so of course it doesn't match.

Any ideas on how to fix this?

**Comment by rainers** [comment]
> Any ideas on how to fix this?

I guess the only way to suppress the output is to redirect it and filter out the filename. Redirecting is also done for Posix to detect some specific error from the linker.

Here's the core function of pipedmd that does something similar i.e. demangling linker error messages on Windows: https://github.com/dlang/visuald/blob/master/tools/pipedmd.d#L252

**Comment by WalterBright** [comment]
@rainers thanks, it's what I need!

**Comment by WalterBright** [comment]
I had to split one of the .c files off into its own .i file, as Microsoft's preprocessor would crash on the ^Z in it.

**Comment by WalterBright** [comment]
@RazvanN7 this is working now. The FreeBSD failure appears to be a networking issue.

File: src/dmd/cpreprocess.d [hunk 1]
```diff
@@ -65,6 +66,21 @@ FileName preprocess(FileName csrcfile)
               there. To get the headers and sppn.exe, dmc will need to be installed on the test machines:
               http://ftp.digitalmars.com/Digital_Mars_C++/Patch/dm857c.zip
          */
+        if (target.objectFormat() == Target.ObjectFormat.coff)
```
Comment (rainers) [inline src/dmd/cpreprocess.d:69]: The comment a couple of lines above is now obsolete.

File: src/dmd/mars.d [hunk 2]
```diff
@@ -732,6 +732,8 @@ bool parseCommandlineAndConfig(size_t argc, const(char)** argv, ref Param params
 
     version (Posix)
         global.preprocess = &preprocess;
+    version (Win64)
```
Comment (rainers) [inline src/dmd/mars.d:735]: You don't have to restrict this to dmd being compiled as a 64-bit program, `version(Windows)` should be fine, too.

Comment (WalterBright) [reply src/dmd/mars.d:735]: I don't want it to try the DMC win32 yet.

File: src/dmd/cpreprocess.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/cpreprocess.d:69]: I'll add sppn support after this is pulled.

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/mars.d (hunk 2 above)
Comment (rainers) [reply src/dmd/mars.d:735]: The version is not about the target tool chain, but the host compiler. The standard dmd build will not execute the preprocessor.

**Pull request merged**